gprofiler/profilers/perf.py

```diff
@@ -107,6 +107,8 @@ def wait_and_script(self) -> str:
             perf_data = wait_for_file_by_prefix(f"{self._output_path}.", self._dump_timeout_s, self._stop_event)
         except Exception:
             assert self._process is not None and self._process.stdout is not None and self._process.stderr is not None
+            os.set_blocking(self._process.stdout.fileno(), False)
+            os.set_blocking(self._process.stderr.fileno(), False)
             logger.critical(
                 f"perf failed to dump output. stdout {self._process.stdout.read()!r}"
```

Comment:
What happens to this `read()` when it's nonblocking? We get back an empty string?

Reply:
We get None

Reply:
kk